Add userscript documentation. (#7488)
GoogleChrome/developer.chrome.com: site/en/docs/extensions/reference/userscript/index.md

---
api: userScript
---

> ## Using now

Reviewer: I really wish this section was an aside 🤔

Author: Generally you only need one way to draw attention to something. This already has the advantage of being close to the top.

This is an experimental feature that is not yet available in Chrome by default. You must take extra steps to enable it for experimentation.

To enable it from a command line, use the following flag with the appropriate [command for your operating system](https://www.chromium.org/developers/how-tos/run-chromium-with-flags/).

`--enable-features="ApiUserScripts"`

## More information and feedback

> You can find more information about this [feature in the explainer](https://github.com/w3c/webextensions/blob/main/proposals/user-scripts-api.md). If you have feedback on the feature or find something that doesn't work, file a bug.

Author: I took the suggestion from the second sentence, but not the one from the first. I'm establishing boilerplate for future use. It should require as little tweaking as possible.

> You can find more information about this [feature in the explainer](https://github.com/w3c/webextensions/blob/main/proposals/user-scripts-api.md). If you have feedback on the feature or find something that doesn't work, file a bug.

Author: ```suggestion
You can find more information about this [feature in the explainer](https://github.com/w3c/webextensions/blob/main/proposals/user-scripts-api.md). If you have any feedback or find something that doesn't work, let us know by filing a bug.
```

## Manifest

> To use the `chrome.userScripts` API, you need to specify a `"manifest_version"` of `3` or higher. The manifest must include the `"userScripts"` permission and `"host_permissions"` for sites you want to run scripts on.

Reviewer: I think you can use the activeTab as well, like in the [Scripting API](https://developer.chrome.com/docs/extensions/reference/scripting/#manifest). I'm confirming with Oliver.

Author: Do you have an answer?

Reviewer: Let's just merge with host permissions. We can double-check the "activeTab" permission later and add it on a separate PR

```json/9-15
{

>   "name": "User script test extension",
>   "version": "0.1",
>   "manifest_version": 3,
>   "description": "Tests a variety of user scripts API calls for registering an user script.",
>   "background": {
>     "service_worker": "service-worker.js"
>   },
>   "minimum_chrome_version": "118",
>   "permissions": [
>     "userScripts"
>   ],
>   "host_permissions": [
>     "*://requested.com/*"
>   ]
> }

Reviewer: ```suggestion
  "name": "User script test extension",
  "manifest_version": 3,
  "minimum_chrome_version": "118",
  "permissions": [
    "userScripts"
  ],
  "host_permissions": [
    "*://requested.com/*"
  ]
}
```
I removed the keys that weren't relevant. You will probably need to update the highlighted items

```

## Concepts and Usage

A user script is a bit of code injected into a web page to modify its appearance or behavior. Scripts are either created by users or downloaded from a script repository using a type of extension called a user script manager.

### Basic usage

> The following example shows a basic call to `register()`. The first argument to register() as an array of objects defining the scripts to be registered. There are more options than are shown here. You can read about them below.

Reviewer: I'm guessing a code sample is coming later 👍

Author: Yes.
